app/src/main/java/org/sil/gatherwords/SessionActivity.java

```diff
@@ -206,4 +233,39 @@ public void onItemSelected(AdapterView<?> parent, View view,
     public void onNothingSelected(AdapterView<?> parent) {
         worldListToLoad = "";
     }
+
+
+    private static class GetInfoFromDB extends AsyncTask<Integer, Void, List<Session>> {
+        private SessionDao sDAO;
+        private WeakReference<SessionActivity> sessionActivityRef;
+
+        GetInfoFromDB(SessionActivity sessionActivity) {
+            sDAO = AppDatabase.get(sessionActivity).sessionDao();
+            sessionActivityRef = new WeakReference<>(sessionActivity);
+        }
+
+        @Override
+        protected List<Session> doInBackground(Integer... ids) {
+            return sDAO.getSessionByID(ids);
+        }
+
+        @Override
+        protected void onPostExecute(List<Session> sessions) {
```

Comment:
Since we are expecting `sessions` to always have exactly one element, perhaps it would be wise to log a warning if it is empty/null or has multiple elements.